tests/publiccloud/ipa.pm

```diff
@@ -15,13 +15,33 @@ use base "opensusebasetest";
 use strict;
 use testapi;
 use utils;
-use File::Basename;
-use registration 'add_suseconnect_product';
 use serial_terminal 'select_virtio_console';
-use version_utils qw(is_sle is_opensuse);
 
+sub get_ipa_tests {
+    my $ipa_tests = get_required_var('PUBLIC_CLOUD_IPA_TESTS');
+    $ipa_tests =~ s/,/ /g;
+    return $ipa_tests;
+}
+
+sub save_logs {
+    my $ret = script_run("test -d ipa_results");
+    return if (!defined($ret) || $ret != 0);
```

Comment:
so you don't want to save logs when you getting non-zero result from test execution ? Am I missing something or you don't want to save logs when something is wrong ?

Reply:
I will check if the "ipa_results" directory exists and only then upload the logs. This directory is created when ipa runs, but the test could fail before.